final-project/scripts/main.js

```diff
@@ -1,10 +1,168 @@
+    if (currentBlog >= length) currentBlog = 0;
+    if (currentBlog < 0) currentBlog = length-1;
+    updateSliderColor(currentBlog);
+    modifyBlogData(currentBlog);
+  }
+  e.stopPropagation();
+}
+
+function updateSliderColor(currentSlider) {
+  var buttons = document.getElementsByClassName("blog-section__slider-btn");
+  for (var i = 0; i < buttons.length; i++) {
+    buttons[i].style.backgroundColor = "#D1D6E3";
+  }
+  buttons[currentSlider].style.backgroundColor = "#5283FF";
+}
+
+function addButtonListenersCustomer(){
+  var buttonsParent = document.querySelector(".customers-section__slider");
+  buttonsParent.addEventListener("click", updateCurrentSlide, false);
+}
+
+function updateCurrentSlide(e) {
+  if (e.target !== e.currentTarget) {
+    var clickedCustomer = e.target.id;
+    currentCustomer(clickedCustomer);
+  }
+  e.stopPropagation();
+}
+
+
+function mediaQueries() {
+  var mobile = window.matchMedia("(max-width: 400px)");
+  screenWidthChangeMobile(mobile);
+  mobile.addListener(screenWidthChangeMobile);
+
+  var tabletPortrait = window.matchMedia("(min-width: 401px) and (max-width: 768px)");
+  screenWidthChangeTabletPortrait(tabletPortrait);
+  tabletPortrait.addListener(screenWidthChangeTabletPortrait);
+
+  var tabletLandscape = window.matchMedia("(min-width: 769px) and (max-width: 1024px)");
+  screenWidthChangeTabletLandscape(tabletLandscape);
+  tabletLandscape.addListener(screenWidthChangeTabletLandscape);
+
+  var desktop = window.matchMedia("(min-width: 1025px)");
+  screenWidthChangeDesktop(desktop);
+  desktop.addListener(screenWidthChangeDesktop);
+}
+
+function screenWidthChangeDesktop(desktop){
+  if(desktop.matches){
+    var x = document.getElementsByClassName("customers-section__images");
+    for (var i = 0; i < x.length; i++) {
+      x[i].style.display = "block";
+    }
+    var image = document.getElementsByClassName("blog-section__content-image");
+    image[0].src = blogData.articles[currentBlog].images.desktop;
+  }
+}
+
+function screenWidthChangeTabletLandscape(tabletLandscape){
+  if (tabletLandscape.matches) {
+    var image = document.getElementsByClassName("blog-section__content-image");
+    image[0].src = blogData.articles[currentBlog].images.tablet;
+  }
+}
+
+function screenWidthChangeTabletPortrait(tabletPortrait) {
+  if (tabletPortrait.matches) {
+    var image = document.getElementsByClassName("blog-section__content-image");
+    image[0].src = blogData.articles[currentBlog].images.mobile2x;
+  }
+}
+
+function screenWidthChangeMobile(mobile) {
+  if (mobile.matches) {
+    displayCustomer(customerInDisplay);
+    var image = document.getElementsByClassName("blog-section__content-image");
+    image[0].src = blogData.articles[currentBlog].images.mobile;
+  }
+}
+
+function currentCustomer(n) {
+  displayCustomer(customerInDisplay = n);
+}
+function displayCustomer(n) {
+  var i;
+  var x = document.getElementsByClassName("customers-section__images");
```

Comment:
Just as a recommendation, when selecting nodes with js is better to do it with selectors that are not related with styles and use a prefix, ie: .js-my-class-selector